spec/lib/administrate/search_spec.rb

```diff
@@ -0,0 +1,59 @@
+require "spec_helper"
+require "support/constant_helpers"
+require "administrate/fields/string"
+require "administrate/fields/email"
+require "administrate/fields/number"
+require "administrate/search"
+
+class User
+end
+
+class MockDashboardClass
+  ATTRIBUTE_TYPES = {
+    name: Administrate::Field::String,
+    email: Administrate::Field::Email,
+    phone: Administrate::Field::Number,
+  }
+end
+
+class MockResolver
+  def resource_class
+    User
+  end
+
+  def dashboard_class
+    MockDashboardClass
+  end
+end
+
+describe Administrate::Search do
+  describe "#run" do
+    let(:resolver) { MockResolver.new }
+
+    it "returns all records when no search term" do
+      search = described_class.new(resolver, nil)
+      expect(User).to receive(:all)
+
+      search.run
+    end
+
+    it "returns all records when search is empty" do
+      search = described_class.new(resolver, '   ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.